### ISSUE TITLE
Rogue code in `setup.py`

### ISSUE THREAD

**Issue by JDBetteridge** [firedrakeproject/Irksome#69]
Why is the following code in `setup.py`?
```python
if "clean" in sys.argv[1:]:
    pass
else:
    try:
        import firedrake # noqa
    except ImportError:
        raise Exception("Firedrake needs to be installed and activated. "
                        "Please visit firedrakeproject.org")
```

It causes `firedrake-update` to always fail as the `firedrake` package is (temporarily) missing whilst Irksome is updated.

If it's 100% necessary to `import firedrake`at install time I'll move the issue to Firedrake, but this is an odd way to check for the dependency.

**Comment by ScottMacLachlan** [comment]
I think there's 2 parts to this:
1) Does `Irksome` need `firedrake` to be available before running `python setup.py install`?  I don't know personally that it does (probably @rckirby has an opinion), so maybe that part could be removed.  As far as I can tell, we've just always done it this way...
2) Why is this wrapped in checking if we're calling `python setup.py clean`?  This was merged in #15.  I put it in because `firedrake-update` expects (or expected at the time) to be able to run `python setup.py clean` on the repo, and that would fail if you couldn't `import firedrake`, which was problematic if you were trying to rebuild things.  If the answer to question 1 above is "no", then this part can certainly go.

**Comment by dham** [comment]
Irksome is a plain Python package. I'm pretty certain Firedrake is only a runtime dependency.

**Comment by rckirby** [comment]
Resolved in #72
